xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/ViewFile.xml

```diff
@@ -263,24 +386,140 @@
+            &lt;span aria-hidden="true"&gt;&amp;times;&lt;/span&gt;&lt;/button&gt;
+          &lt;div id="${modalId}Label" class="modal-title"&gt;
+            $escapetool.xml($services.localization.render('rendering.macro.viewFiles.modal.title'))
+          &lt;/div&gt;
+        &lt;/div&gt;
+        &lt;div class="modal-body"&gt;
+
+        {{office reference="attach:$escapedFilename" /}}
+
+        &lt;/div&gt;
+        &lt;div class="modal-footer"&gt;
+          &lt;button type="button" class="btn btn-default" data-dismiss="modal"&gt;
+            $escapetool.xml($services.localization.render('cancel'))&lt;/button&gt;
+        &lt;/div&gt;
+      &lt;/div&gt;
+    &lt;/div&gt;
+  &lt;/div&gt;
+#end
 #macro (executeMacro)
+  #set ($discard = $xwiki.ssx.use('Confluence.Macros.ViewFile'))
   #set($hasPDFViewer = $xwiki.exists("XWiki.PDFViewerMacro"))
   #set($officeExtensions = [ 'ppt', 'pptx', 'odp', 'doc', 'docx', 'odt', 'xls', 'xlsx', 'ods' ])
+  #set($presentationExtensions = [ 'ppt', 'pptx', 'odp'])
   #set($unescapedFilename = $xcontext.macro.params.get('att--filename'))
   #if(!$unescapedFilename)
     #set($unescapedFilename = $xcontext.macro.params.get('name'))
   #end
   #set($extension = $unescapedFilename.substring($mathtool.add($unescapedFilename.lastIndexOf('.'), 1)).toLowerCase())
   #set($escapedFilename = $services.rendering.escape($unescapedFilename, $xwiki.currentContentSyntaxId))
-  #if($extension == 'pdf' &amp;&amp; $hasPDFViewer)
+  #set($width = $xcontext.macro.params.get('width'))
+  #set($height = $xcontext.macro.params.get('height'))
+  #set($display = $xcontext.macro.params.get('display'))
+  #set($idName = $unescapedFilename.replaceAll("[^a-zA-Z]", ""))
+  {{html wiki="true" clean="false"}}
+    #if ($display == 'FULL')
+      #if ($presentationExtensions.contains($extension))
+        #if($height &gt; 390)
+          #set($height = '390')
+        #end
 
-    {{pdfviewer file="$escapedFilename" /}}
-  #elseif($officeExtensions.contains($extension))
+        #customPresentationBox($width $height $escapedFilename)
+      #elseif($extension == 'pdf' &amp;&amp; $hasPDFViewer)
 
-    {{office reference="attach:$escapedFilename" /}}
-  #elseif($doc.getAttachment($unescapedFilename))
-    [[attach:$escapedFilename]]
-  #end
+        {{pdfviewer file="$escapedFilename" height=$height width=$width/}}
+      #elseif($officeExtensions.contains($extension))
+
+        #customBox($width $height $escapedFilename)
+      #elseif($doc.getAttachment($unescapedFilename))
+        #set ($downloadUrl = $xwiki.getAttachmentURL($doc.getFullName(), $unescapedFilename))
+        &lt;div&gt;
+          &lt;a href="$downloadUrl" download&gt;$escapedFilename&lt;br&gt;&lt;/a&gt;
+        &lt;/div&gt;
+      #end
+    #elseif ($display == 'THUMBNAIL')
+      #if (($extension == 'pdf' &amp;&amp; $hasPDFViewer) || $officeExtensions.contains($extension))
+        #set ($modalId = $idName + 'DisplayContentModal')
```

Comment:
You should escape the modal id in all the places that u use it to generate html.